src/Htmxor/Endpoints/Results/HtmxComponentResultExecutor.cs

```diff
@@ -0,0 +1,92 @@
+
+	public static Task ExecuteAsync(HttpContext httpContext, HtmxorComponentResult result)
+	{
+		ArgumentNullException.ThrowIfNull(httpContext);
+
+		var response = httpContext.Response;
+		response.ContentType = result.ContentType ?? DefaultContentType;
+
+		if (result.StatusCode != null)
+		{
+			response.StatusCode = result.StatusCode.Value;
+		}
+
+		return RenderComponentToResponse(
+			httpContext,
+			result.ComponentType,
+			result.Parameters,
+			result.PreventStreamingRendering);
+	}
+
+	private static Task RenderComponentToResponse(
+		HttpContext httpContext,
+		[DynamicallyAccessedMembers(Component)] Type componentType,
+		IReadOnlyDictionary<string, object?>? componentParameters,
+		bool preventStreamingRendering)
+	{
+		var endpointHtmlRenderer = httpContext.RequestServices.GetRequiredService<HtmxorRenderer>();
+		return endpointHtmlRenderer.Dispatcher.InvokeAsync(async () =>
+		{
+			var isErrorHandler = httpContext.Features.Get<IExceptionHandlerFeature>() is not null;
+
+			//endpointHtmlRenderer.InitializeStreamingRenderingFraming(httpContext, isErrorHandler);
+
+			// We could pool these dictionary instances if we wanted, and possibly even the ParameterView
+			// backing buffers could come from a pool like they do during rendering.
+			var hostParameters = ParameterView.FromDictionary(new Dictionary<string, object?>
+			{
+				{ nameof(HtmxorComponentEndpointHost.ComponentType), componentType },
+				{ nameof(HtmxorComponentEndpointHost.ComponentParameters), componentParameters },
+			});
+
+			// Matches MVC's MemoryPoolHttpResponseStreamWriterFactory.DefaultBufferSize
+			var defaultBufferSize = 16 * 1024;
+			await using var writer = new HttpResponseStreamWriter(httpContext.Response.Body, Encoding.UTF8, defaultBufferSize, ArrayPool<byte>.Shared, ArrayPool<char>.Shared);
+			using var bufferWriter = new BufferedTextWriter(writer);
+
+			// Note that we don't set any interactive rendering mode for the top-level output from a RazorComponentResult,
+			// because you never want to serialize the invocation of RazorComponentResultHost. Instead, that host
+			// component takes care of switching into your desired render mode when it produces its own output.
+			var htmlContent = await endpointHtmlRenderer.RenderEndpointComponent(httpContext, typeof(HtmxorComponentEndpointHost), hostParameters);
+			
+			// Importantly, we must not yield this thread (which holds exclusive access to the renderer sync context)
+			// in between the first call to htmlContent.WriteTo and the point where we start listening for subsequent
+			// streaming SSR batches (inside SendStreamingUpdatesAsync). Otherwise some other code might dispatch to the
+			// renderer sync context and cause a batch that would get missed.
+#pragma warning disable CA1849
+			htmlContent.WriteTo(bufferWriter, HtmlEncoder.Default); // Don't use WriteToAsync, as per the comment above
+#pragma warning restore CA1849
```

Comment:
This should probably look like the code in `HtmxorComponentEndpointInvoker`, i.e.:

```suggestion
			// Note that we don't set any interactive rendering mode for the top-level output from a RazorComponentResult,
			// because you never want to serialize the invocation of RazorComponentResultHost. Instead, that host
			// component takes care of switching into your desired render mode when it produces its own output.
			var htmlContent = await endpointHtmlRenderer.RenderEndpointComponent(httpContext, typeof(HtmxorComponentEndpointHost), hostParameters);
			
			if (context.Response.StatusCode == (int)HttpStatusCode.NoContent || htmxContext.Response.EmptyResponseBodyRequested)
			{
				return;
			}
			
			var defaultBufferSize = 16 * 1024;
			await using var writer = new HttpResponseStreamWriter(httpContext.Response.Body, Encoding.UTF8, defaultBufferSize, ArrayPool<byte>.Shared, ArrayPool<char>.Shared);
			using var bufferWriter = new BufferedTextWriter(writer);
			
			// Importantly, we must not yield this thread (which holds exclusive access to the renderer sync context)
			// in between the first call to htmlContent.WriteTo and the point where we start listening for subsequent
			// streaming SSR batches (inside SendStreamingUpdatesAsync). Otherwise some other code might dispatch to the
			// renderer sync context and cause a batch that would get missed.
#pragma warning disable CA1849
			htmlContent.WriteTo(bufferWriter, HtmlEncoder.Default); // Don't use WriteToAsync, as per the comment above
#pragma warning restore CA1849
```

Reply:
Implemented these changes